pkg/vscrstatus/status.go

```diff
@@ -92,3 +95,17 @@ func UpdateCondition(ctx context.Context, clnt client.Client, vscr *v1beta1.Vert
 
 	return update(ctx, clnt, vscr, refreshConditionInPlace)
 }
+
+// UpdateState will update the state field in vscr status
+func UpdateState(ctx context.Context, clnt client.Client, vscr *v1beta1.VerticaScrutinize, state string) error {
+	// refreshState will update the state field in vscr.  The update
+	// will be applied in-place.
+	refreshState := func(vscr *v1beta1.VerticaScrutinize) error {
```

Comment:
I noticed that none of the callback functions we pass to `update` return an error. They all return `nil`. Could we change the callback function signature to not return an error?